### PR TITLE
EQL: Allow requests with size 0

### LINKED ISSUE #62494: EQL: Support size 0 for query validation
As part of the effort of building out our new EQL Rule type for Kibana's Security Detection Engine, we want validation of EQL queries as users build their rules. Rather than build a dedicated endpoint for validation, it was suggested that we perform a query of `size: 0` to receive any `parsing_exception`s etc. but not incur the cost of data retrieval/processing.

However, providing `size: 0` to an EQL query currently returns the following error: 

```json
{
  "error" : {
    "root_cause" : [
      {
        "type" : "action_request_validation_exception",
        "reason" : "Validation Failed: 1: size must be greater than 0;"
      }
    ],
    "type" : "action_request_validation_exception",
    "reason" : "Validation Failed: 1: size must be greater than 0;"
  },
  "status" : 400
}
```

While we can use `size: 1` for the time being and incur that additional overhead, support for `size: 0` would be great!

### PR DESCRIPTION
**Pull request by costin**
The purpose for this change is to allow validation of queries without
having to actually execute them. The optimizer already picks up this
case.

Fix #62494

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/eql/EqlSearchRequest.java [hunk 1]
```diff
@@ -158,8 +158,8 @@ public int size() {
 
     public EqlSearchRequest size(int size) {
         this.size = size;
-        if (fetchSize <= 0) {
-            throw new IllegalArgumentException("size must be greater than 0");
+        if (fetchSize < 0) {
```
Comment (astefan) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/eql/EqlSearchRequest.java:161]: Shouldn't this be `size` and not `fetchSize`?

File: client/rest-high-level/src/test/java/org/elasticsearch/client/eql/EqlSearchRequestTests.java [hunk 2]
```diff
@@ -35,30 +35,33 @@
...
     protected EqlSearchRequest createClientTestInstance() {
-        EqlSearchRequest EqlSearchRequest = new EqlSearchRequest("testindex", randomAlphaOfLength(40));
+        EqlSearchRequest eqlSearchRequest = new EqlSearchRequest("testindex", randomAlphaOfLength(40));
         if (randomBoolean()) {
-            EqlSearchRequest.fetchSize(randomIntBetween(1, Integer.MAX_VALUE));
+            eqlSearchRequest.fetchSize(randomIntBetween(1, Integer.MAX_VALUE));
         }
         if (randomBoolean()) {
-            EqlSearchRequest.eventCategoryField(randomAlphaOfLength(10));
+            eqlSearchRequest.size(randomInt(Integer.MAX_VALUE));
```
Comment (astefan) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/eql/EqlSearchRequestTests.java:43]: Shouldn't this be `randomIntBetween(0, Integer.MAX_VALUE)`? or maybe to cover better the 0 scenario - a `randomBoolean()` to choose between `size` 0 and `size` between 1 and MAX.

**Review by astefan — COMMENTED** [review]
Left couple of comments.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/eql/EqlSearchRequest.java (hunk 1 above)
Comment (costin) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/eql/EqlSearchRequest.java:161]: 🤦

File: client/rest-high-level/src/test/java/org/elasticsearch/client/eql/EqlSearchRequestTests.java (hunk 2 above)
Comment (costin) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/eql/EqlSearchRequestTests.java:43]: `randomInt` is `randomIntBetween(0, Integer.MAX_VALUE)`

**Review by bpintea — APPROVED** [review]

**Review by matriv — APPROVED** [review]
LGTM

**Pull request merged**